crates/sui-indexer-alt/src/handlers/wal_coin_balances.rs

```diff
@@ -56,7 +56,8 @@ impl Handler for WalCoinBalances {
             .await?)
     }
 
-    async fn prune(from: u64, to: u64, conn: &mut db::Connection<'_>) -> Result<usize> {
+    async fn prune(range: PrunableRange, conn: &mut db::Connection<'_>) -> Result<usize> {
+        let (from, to) = range.checkpoint_interval();
```

Comment:
So one thing I'm noting here is that:

- This pipeline doesn't need to fetch the `PrunableRange` in order to get its checkpoint interval.
- Because the `prune` function has access to a `Connection`, it's perfectly capable of converting a pair of checkpoints into a `PrunableRange` if it needs to.

Does it make sense to keep the interface to `prune` checkpoint based and then allow the `prune` implementation to fetch the range if it needs?

Reply:
Your suggestion sounds pretty good to me. This would also no longer be a breaking change, as the two existing pipelines can continue to directly make use of the `from` and `to` checkpoints, while other implementations can leverage `PrunableRange` accordingly.